app/models/concerns/searchable.rb

```diff
@@ -12,7 +12,7 @@ def index_to_elasticsearch_inline
   end
 
   def remove_from_elasticsearch
-    Search::RemoveFromIndexWorker.perform_async(self.class::SEARCH_CLASS.to_s, search_id)
+    Search::RemoveFromIndexWorker.perform_in(5.seconds, self.class::SEARCH_CLASS.to_s, search_id)
```

Comment:
Not a block but it would be a good idea to leave a comment that this is to avoid that edge-case.

Reply:
good call!